browser/sources.gni

```diff
@@ -576,6 +577,12 @@ brave_chrome_browser_deps += brave_browser_webcompat_reporter_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_deps
 brave_chrome_browser_deps += brave_chromium_src_chrome_browser_prefs_deps
 
+brave_chrome_browser_ui_allow_circular_includes_from = [ "//brave/browser/ui" ]
+if (!is_android) {
+  brave_chrome_browser_ui_allow_circular_includes_from +=
+      [ "//brave/browser/ui/webui/brave_news_internals" ]
+}
```

Comment:
Hey, so this PR looks good to me, but I'm curious about why we prefer allowing circular includes here, rather than adding the files to the target like everything else does?

Reply:
because adding everything to one target is really bad. We have been working on fixing this and so is upstream. 

https://source.chromium.org/chromium/chromium/src/+/main:docs/chrome_browser_design_principles.md;l=25

Reply:
Ah cool, I didn't realise upstream was working to fix this! That's going to be super handy

Reply:
That's super exciting actually - its always a bit demoralizing when you have an almost-completely-independent target and then have to add it to one of the big ones for some new dep!